src/Microsoft.ML.Transforms/CustomMappingTransformer.cs

```diff
@@ -58,6 +60,7 @@ public sealed class CustomMappingTransformer<TSrc, TDst> : ITransformer
                : InternalSchemaDefinition.Create(typeof(TDst), outputSchemaDefinition);
 
             _contractName = contractName;
+            _contractAssembly = _mapAction.Method.DeclaringType.Assembly.FullName;
```

Comment:
Is there any case where the loaded model would actually require having a different name registered from the "FullName" retrieved from here?

Reply:
Or any case where trying to access that member of `_mapAction` would throw?

Reply:
The call to Method can throw a MemberAccessException. But that would be up to the caller to fix in their code and the exception would help with that. 

---
In reply to: [401790783](https://github.com/dotnet/machinelearning/pull/4989#discussion_r401790783) [](ancestors = 401790783)

Reply:
I think we should enforce the idea that the same transform that is used in training should be used in prediction as well. If they were to be different, then they are not the same pipelines and not the same models. 

---
In reply to: [401431899](https://github.com/dotnet/machinelearning/pull/4989#discussion_r401431899) [](ancestors = 401431899)